shipyard-controller/db/mongodb_sequence_execution_repo_test.go

```diff
@@ -0,0 +1,260 @@
+}
+
+func TestMongoDBTaskSequenceV2Repo_GetByTriggeredID(t *testing.T) {
+	scope, sequence := getTestSequenceExecution()
+
+	mdbrepo := NewMongoDBSequenceExecutionRepo(GetMongoDBConnectionInstance())
+
+	err := mdbrepo.Upsert(sequence, nil)
+
+	require.Nil(t, err)
+
+	get, err := mdbrepo.Get(models.SequenceExecutionFilter{
+		Scope:  scope,
+		Name:   "delivery",
+		Status: []string{"triggered"},
+	})
+
+	require.Nil(t, err)
+
+	require.Len(t, get, 1)
+	require.Equal(t, sequence, get[0])
+
+	sequenceByTriggeredID, err := mdbrepo.GetByTriggeredID("my-project", "my-triggered-id")
+
+	require.Nil(t, err)
+	require.NotNil(t, sequenceByTriggeredID)
+}
+
+func TestMongoDBTaskSequenceV2Repo_InsertTwice(t *testing.T) {
+	_, sequence := getTestSequenceExecution()
+
+	mdbrepo := NewMongoDBSequenceExecutionRepo(GetMongoDBConnectionInstance())
+
+	err := mdbrepo.Upsert(sequence, nil)
+
+	require.Nil(t, err)
+
+	// try to insert the same sequence again, but with check for already existing triggeredID - this should return an error
+	err = mdbrepo.Upsert(sequence, &models.SequenceExecutionUpsertOptions{CheckUniqueTriggeredID: true})
+
+	require.ErrorIs(t, err, ErrSequenceWithTriggeredIDAlreadyExists)
+}
+
+func TestMongoDBTaskSequenceV2Repo_AppendTaskEvent(t *testing.T) {
+	scope, sequence := getTestSequenceExecution()
+
+	mdbrepo := NewMongoDBSequenceExecutionRepo(GetMongoDBConnectionInstance())
+
+	err := mdbrepo.Upsert(sequence, nil)
+
+	require.Nil(t, err)
+
+	get, err := mdbrepo.Get(models.SequenceExecutionFilter{
+		Scope:  scope,
+		Name:   "delivery",
+		Status: []string{"triggered"},
+	})
+
+	require.Nil(t, err)
+
+	require.Len(t, get, 1)
+	require.Equal(t, sequence, get[0])
+
+	triggeredEvent := models.TaskEvent{
+		EventType: "deploy.triggered",
+		Source:    "my-source",
+		Time:      timeutils.GetKeptnTimeStamp(time.Now().UTC()),
+	}
+	result, err := mdbrepo.AppendTaskEvent(get[0], triggeredEvent)
+
+	require.Nil(t, err)
+
+	require.Len(t, result.Status.CurrentTask.Events, 1)
+	require.Equal(t, triggeredEvent, result.Status.CurrentTask.Events[0])
+}
+
+func TestMongoDBTaskSequenceV2Repo_AppendTaskEventMultipleWriters(t *testing.T) {
+	scope, sequence := getTestSequenceExecution()
+
+	mdbrepo := NewMongoDBSequenceExecutionRepo(GetMongoDBConnectionInstance())
+
+	err := mdbrepo.Upsert(sequence, nil)
+
+	require.Nil(t, err)
+
+	get, err := mdbrepo.Get(models.SequenceExecutionFilter{
+		Scope:  scope,
+		Name:   "delivery",
+		Status: []string{"triggered"},
+	})
+
+	require.Nil(t, err)
+
+	require.Len(t, get, 1)
+	require.Equal(t, sequence, get[0])
+
+	triggeredEvent := models.TaskEvent{
+		EventType: "deploy.triggered",
+		Source:    "my-source",
+		Time:      timeutils.GetKeptnTimeStamp(time.Now().UTC()),
+	}
+
+	// ensure that multiple writers can append data to a shared sequence and all inserts are persisted
+	nrConcurrentWrites := 100
+
+	wg := sync.WaitGroup{}
+
+	wg.Add(nrConcurrentWrites)
+
+	for i := 0; i < nrConcurrentWrites; i++ {
```

Comment:
this test makes sure that the `AppendTaskEvent` works in a thread safe manner - i.e. the threads invoking this function should not overwrite each other's changes